common/workunit/workunit.cpp

```diff
@@ -4611,6 +4611,34 @@ IStringVal &getProcessQueueNames(IStringVal &ret, const char *process, const cha
 #define ECLSCHEDULER_QUEUE_EXT ".eclscheduler"
 #define ECLAGENT_QUEUE_EXT ".agent"
 
+extern WORKUNIT_API IStringVal &getDFUServerQueueNames(IStringVal &ret, const char *process, const char *propName)
+{
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
+    Owned<IConstEnvironment> env = factory->openEnvironment();
+    if (!env)
+        return ret;
+
+    VStringBuffer pName("@%s", (propName && *propName) ? propName: "queue");
+    StringBuffer xpath = "Software/DfuServerProcess";
+    if (process && *process)
+        xpath.appendf("[@name=\"%s\"]", process);
+
+    StringBuffer queueNames;
+    Owned<IPropertyTree> root = &env->getPTree();
+    Owned<IPropertyTreeIterator> targets = root->getElements(xpath.str());
+    ForEach(*targets)
+    {
+        IPropertyTree &target = targets->query();
+        if (!target.hasProp(pName.str()))
+            continue;
+        if (queueNames.length())
+            queueNames.append(',');
```

Comment:
Would be cleaner if used StringArray and appendList .

And should it dedup them? e.g. use appendListUnique